Source/WebCore/animation/ScrollTimeline.cpp

```diff
@@ -138,4 +140,35 @@ AnimationTimeline::ShouldUpdateAnimationsAndSendEvents ScrollTimeline::documentW
     return AnimationTimeline::ShouldUpdateAnimationsAndSendEvents::No;
 }
 
+ScrollTimeline::ScrollTimelineData ScrollTimeline::computeScrollTimelineData()
+{
+    if (!m_source)
+        return { };
+
+    CheckedPtr renderBox = dynamicDowncast<RenderBox>(m_source->renderer());
+    if (!renderBox)
+        return { };
+
+    // https://drafts.csswg.org/scroll-animations-1/#scroll-timeline-progress
+    // Progress (the current time) for a scroll progress timeline is calculated as:
+    // scroll offset ÷ (scrollable overflow size − scroll container size)
+    float scrollOffset = 0;
+    float scrollableOverflowSize;
+    float scrollContainerSize;
+
+    if (Ref { m_source->document() }->scrollingElement() == m_source.get()) {
```

Comment:
This is repeated from the ViewTimeline PR. Let's share code.

Reply:
Currently I think the ViewTimeline code should call into this to get the data for the source element.